platforms/software/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/AbstractInitIntegrationSpec.groovy

```diff
@@ -29,14 +32,19 @@ import static org.gradle.initialization.StartParameterBuildOptions.Configuration
 import static org.hamcrest.MatcherAssert.assertThat
 import static org.hamcrest.Matchers.containsString
 import static org.hamcrest.Matchers.not
+import org.gradle.api.internal.tasks.testing.report.VerifiesGenericTestReportResults
 
-abstract class AbstractInitIntegrationSpec extends AbstractIntegrationSpec {
+abstract class AbstractInitIntegrationSpec extends AbstractIntegrationSpec implements VerifiesGenericTestReportResults {
     TestFile containerDir
     TestFile targetDir
     TestFile subprojectDir
 
     abstract String subprojectName()
 
+    TestFramework getTestFramework() {
```

Comment:
This reads weird in the integration tests. I'd propose to turn this into a field, set the initial value in `AbstractInitIntegrationSpec.setup()` and add a new testFramework(TestFramework) method so that the value can be overwritten in the `setup` methods of the classes using a custom test framework.

Reply:
Please take a look at https://github.com/gradle/gradle/pull/32672/commits/3d1db116506ec3a52d8956261e715b2e490ab917.